tracker/classify.go

```diff
@@ -0,0 +1,42 @@
+// Package tracker implements an object tracker as a Viam vision service
+// This file contains methods that are useful for classifying the detections
+package tracker
+
+import (
+	"context"
+	"image"
+
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/services/vision"
+	objdet "go.viam.com/rdk/vision/objectdetection"
+)
+
+// method will take a slice of tracks and return a slice of tracks.
+// The only difference is that the track will now include the detection classification
+func classifyTracks(ctx context.Context, tracks []*track, img image.Image, classifier vision.Service) []*track {
+	
+	for _, tr := range tracks {
+		cropped := cropImageFromDet(img, tr.Det)
+		out, err := classifier.Classifications(ctx, cropped, 1, nil)
+		if err != nil || len(out) > 1 {
+			// if there is an error, just skip the classification
+			continue
+		}
+		tr.detClassification = out[0]
+	}
+
+	return tracks
+}
+
+func cropImageFromDet(img image.Image, det objdet.Detection) image.Image {
+	lazyImg, ok := img.(*rimage.LazyEncodedImage)
+	if ok {
+		im := lazyImg.DecodedImage()
+		cropped := im.(interface {
+			SubImage(r image.Rectangle) image.Image
+		}).SubImage(*det.BoundingBox())
+		return cropped
+	}
+
+	return nil
+}
```

Comment:
This only works for LazyEncodedImage which is pretty limited. We actually also made a crop function in queue-sensor to crop out locations within the arena that works for any image.Image (including Lazy). [It is here](https://github.com/viam-modules/queue-estimator/blob/da9f7c9bbcd5823c1f30f7026d646fa4502ebdf1/waitsensor/waitsensor.go#L160)